src/main.rs

```diff
@@ -79,10 +104,34 @@ fn create_and_add_row(path: String, database: &Connection) -> anyhow::Result<Str
     add_stmt.execute([&uuid, &path, &time])?;
     Ok(uuid)
 }
-async fn redirect(State(shared_state): State<AppState>, data_string: String) -> Response {
+
+async fn redirect_post_mulipart_form(
+    State(shared_state): State<AppState>,
+    mut multipart: Multipart,
+) -> Response {
     let database: tokio::sync::MutexGuard<'_, Connection> = shared_state.database.lock().await;
+    let mut data_array: Vec<File> = Vec::new();
+
+    while let Some(field) = multipart.next_field().await.unwrap() {//remove unwrap()
+        let file_type = field.content_type().unwrap().to_string();
+        let file_type = FileType::from_str(file_type.as_str()).unwrap();
+        let file_name = field.file_name().unwrap().to_string();
 
-    let path = match migrate(data_string) {
+        let data = field.bytes().await.unwrap();
+
+        let file_content = match str::from_utf8(&data) {
+            Ok(v) => v.to_string(),
+            Err(e) => panic!("Invalid UTF-8 sequence: {}", e),
+        };
+
+        data_array.push(File {
+            file_content,
+            file_name,
+            file_type,
+        });
+    }
+
```

Comment:
Now you aren't checking anymore if mixing of ifcfg and xml happens. You can pretty easily compare ensure only the same file types are in the array via the `windows()` and `all()` functions.